src/ocm/components/clusterConfiguration/staticIp/data/formDataToInfraEnvField.ts

```diff
@@ -74,36 +75,28 @@ const getNmstateObject = (
         ),
       };
       nicName = getDummyNicName(protocolVersion);
-      interfaces.push(getEthernetInterface(nicName, protocolConfigs));
+      interfaces.push(getInterface(nicName, protocolConfigs, networkWide));
     }
+
     routeConfigs.push(
-      getRouteConfig(protocolVersion, networkWide.ipConfigs[protocolVersion].gateway, nicName),
+      getRouteConfig(
+        protocolVersion,
+        networkWide.ipConfigs[protocolVersion].gateway,
+        networkWide.useVlan && networkWide.vlanId ? `${nicName}.${networkWide.vlanId}` : nicName,
```

Comment:
One last polish? :)
The code to generate the vlan name repeats itself also in the function getInterface
Would be nice to have it in a function, something like getVlanNicName

Reply:
Done :slightly_smiling_face: